service/worker/archiver/client.go

```diff
@@ -66,18 +70,28 @@ func NewClient(
 	logger log.Logger,
 	publicClient workflowserviceclient.Interface,
 	numWorkflows dynamicconfig.IntPropertyFn,
+	requestRPS int,
 ) Client {
 	return &client{
 		metricsClient: metricsClient,
 		logger:        logger,
 		cadenceClient: cclient.NewClient(publicClient, common.SystemDomainName, &cclient.Options{}),
 		numWorkflows:  numWorkflows,
+		rateLimiter:   tokenbucket.New(requestRPS, clock.NewRealTimeSource()),
 	}
 }
 
 // Archive starts an archival task
 func (c *client) Archive(request *ArchiveRequest) error {
 	c.metricsClient.IncCounter(metrics.ArchiverClientScope, metrics.CadenceRequests)
+
+	if ok, _ := c.rateLimiter.TryConsume(1); !ok {
+		errMsg := "Too many archive requests to archival workflow"
```

Comment:
declare the error message once at the top of the file and reuse it across calls

Reply:
Done.